### PR TITLE
[ashell] Fix add_requires to allow single as well as double quotes

### PR DESCRIPTION
**Pull request by grgustaf**
Also, improve parsing of require() by requiring parentheses, requiring
the string to be found within them, but allowing whitespace around them.

Note some of the remaining foibles with comments.

Signed-off-by: Geoff Gustafson <geoff@linux.intel.com>

### CODE REVIEW

**Comment by grgustaf** [comment]
You can test this just by running `require('gpio');` in the IDE.

**Comment by jimmy-huang** [comment]
+1

File: src/ashell/jerry-code.c [hunk 1]
```diff
@@ -195,53 +195,122 @@ static void eval_list(requires_list_t **list)
...
+    // modifies: *ptr
+    //  effects: skips any whitespace in *ptr before and after a match char or
+    //             until null terminator is reached
+    //           if first non-whitespace char found is match, returns true;
+    //             otherwise, false
+    skip_whitespace(ptr);
+    if (**ptr != match) {
+        return false;
+    }
+    skip_whitespace(ptr);
```
Comment (brianjjones) [inline src/ashell/jerry-code.c]: You need a ++(*ptr); before this line.  Otherwise skip_whitespace won't do anything as ptr is currently pointing at '('

Comment (grgustaf) [inline src/ashell/jerry-code.c]: Thanks!

**Comment by brianjjones** [comment]
@grgustaf Please hold off on this one, I think I have a more elegant solution that doesn't require an extra function.  Will show you first thing tomorrow.

Thanks.

**Pull request merged**